src/main/java/org/broadinstitute/hellbender/utils/SequenceDictionaryUtils.java

```diff
@@ -308,21 +318,32 @@ private static boolean commonContigsHaveSameLengths(Set<String> commonContigs, S
 
     /**
      * Helper routine that returns whether two sequence records are equivalent, defined as having the same name and
-     * lengths
+     * lengths.
+     *
+     * NOTE: we allow the lengths to differ if one or both are UNKNOWN_SEQUENCE_LENGTH
      *
      * @param first first sequence record to compare
      * @param second second sequence record to compare
      * @return true if first and second have the same names and lengths, otherwise false
      */
-    private static boolean sequenceRecordsAreEquivalent(final SAMSequenceRecord first, final SAMSequenceRecord second) {
+    @VisibleForTesting
+    static boolean sequenceRecordsAreEquivalent(final SAMSequenceRecord first, final SAMSequenceRecord second) {
         if ( first == second ) {
             return true;
         }
         if ( first == null || second == null ) {
             return false;
         }
+        if (! first.getSequenceName().equals(second.getSequenceName())){
```

Comment:
Compare the lengths before the names, since comparing the lengths is cheaper than String comparison.